### PR TITLE
Make push work

### PR DESCRIPTION
**Pull request by bcavagnolo**
Hoping that @moore will take a look at this one. A couple of thoughts/questions:

1. Two files with the same name and origin that were generated on the same day will collide in the url scheme even if they represent different timeframes. I considered addressing this by adding some of the metadata fields to the id hashing. But then we can't checksum the file independent of the metadata. I guess I could distinguish between what I call the "id" and the "checksum." Thoughts?

2. The default 64-byte digest feel kind of obnoxious. Any objection if I chop it down to 20?

### CODE REVIEW

File: datalake/archive.py [hunk 1]
```diff
@@ -52,10 +52,19 @@ def _s3_bucket_name(self):
     def _s3_bucket(self):
         return self._s3_conn.get_bucket(self._s3_bucket_name)
 
-    _KEY_FORMAT = '{where}/{what}/{name}'
+    _KEY_FORMAT = '{hash}-{who}/{where}/{what}/{when}/{name}'
```
Comment (moore) [inline datalake/archive.py]: My initial suggestion would be {when}/{name}-{hash}

File: README.md [hunk 2]
```diff
@@ -75,6 +76,7 @@ with each file. In JSON, the metadata looks something like this:
             "end": "2015-03-20T23:59:59.114Z",
             "where": "webserver02",
```
Comment (moore) [inline README.md]: Should include the program that created it with it's version? That might be captured in the data version thou.

File: README.md [hunk 3]
```diff
@@ -75,6 +76,7 @@ with each file. In JSON, the metadata looks something like this:
...
             "what": "syslog",
+            "who": "webteam",
```
Comment (moore) [inline README.md]: As we discussed I don't think this is appropriate as it dose not describe the data.

File: README.md [hunk 4]
```diff
@@ -75,6 +76,7 @@ with each file. In JSON, the metadata looks something like this:
...
             "data-version": "0",
             "tags": {
```
Comment (moore) [inline README.md]: What are these tags for? Why is the who not a tag?

**Comment by bcavagnolo** [comment]
Okay, so, I'm going to break this up into a few separate PRs. On the matter of "who", I've come to agree with @moore. On the matter of URL format, I've accepted that we should not necessarily rely on the URL format for indexing, and that we will need a faster metadata index earlier than I anticipated. That said, I do believe that the value of legible URLs for debugging and browsability outweighs the potential costs (e.g., developers inappropriately relying on the URL format). So I'll probably make the URL format richer than @moore advises.

**Pull request closed without merging**